File: hbTapi Components/BCB Samples/Standard Edition/TranslateAddress/uMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "uMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "hbTapi"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}

//---------------------------------------------------------------------------

void __fastcall TForm1::FormCreate(TObject *Sender)
{
  ComboBox1->Items->Assign(hbTapiLine1->DeviceList);

}

//---------------------------------------------------------------------------

void __fastcall TForm1::Button5Click(TObject *Sender)
{
  if (hbTapiLine1->Available)
    MessageDlg("Dialable address: " + hbTapiLine1->TranslateAddress(Edit1->Text,0), mtInformation, TMsgDlgButtons() << mbOK, 0);
  else
    MessageDlg("No device selected/available", mtError, TMsgDlgButtons() << mbOK, 0);

}
//---------------------------------------------------------------------------


void __fastcall TForm1::ComboBox1Change(TObject *Sender)
{
  hbTapiLine1->DeviceID = ComboBox1->ItemIndex;

}
//---------------------------------------------------------------------------

File: hbTapi Components/Delphi Samples/Standard Edition/TranslateAddress/uMain.pas
{-------------------------------------------------------------------------
   Copyright by Haeger + Busch, Germany                / >>>>>>>>> /-----
   Ingenieurbuero fuer Kommunikationslösungen         / <<<<<<<<< /
 ----------------------------------------------------/ >>>>>>>>> /
   Homepage  : http://www.hbTapi.com
   EMail     : info@hbTapi.com
   Package   : hbTapi Components
 -------------------------------------------------------------------------}

unit uMain;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs, hbtapi,
  StdCtrls, ExtCtrls;

type
  TForm1 = class(TForm)
    hbTapiLine1: ThbTapiLine;
    GroupBox3: TGroupBox;
    Edit1: TEdit;
    Button5: TButton;
    ComboBox1: TComboBox;
    Label1: TLabel;
    Label2: TLabel;
    procedure FormCreate(Sender: TObject);
    procedure Button5Click(Sender: TObject);
    procedure ComboBox1Change(Sender: TObject);
  private
  public
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.FormCreate(Sender: TObject);
begin
  ComboBox1.Items.Assign(hbTapiLine1.DeviceList);
end;

procedure TForm1.Button5Click(Sender: TObject);
begin
  if hbTapiLine1.Available then
    MessageDlg('Dialable address: ' + hbTapiLine1.TranslateAddress(Edit1.Text), mtInformation, [mbOK], 0)
  else
    MessageDlg('No device selected/available', mtError, [mbOK], 0);
end;

procedure TForm1.ComboBox1Change(Sender: TObject);
begin
  hbTapiLine1.DeviceID := ComboBox1.ItemIndex;
end;

end.
